### PR TITLE
Configure QBO's logging with two new settings.

### PR DESCRIPTION
**Pull request by ananelson**
Fix for issue #11 

Adds two new settings and uses them to configure QBO's logging behavior.

### CODE REVIEW

File: settings/civiquickbooks.setting.php [hunk 1]
```diff
@@ -139,6 +139,39 @@
     ),
     'quick_form_type' => 'Element',
   ),
+  'quickbooks_log_dir' => array(
+    'group_name' => E::ts('QuickBooks Online Settings'),
+    'group' => 'civiquickbooks',
+    'name' => 'quickbooks_log_dir',
+    'type' => 'String',
+    'add' => '2.7', // this is the civiquickboks version number where this was added
```
Comment (ananelson) [inline settings/civiquickbooks.setting.php:147]: Having trouble finding unambiguous documentation for what this 'add' parameter is supposed to represent. A version number, yes, but version for what? This is my best guess.

File: CRM/Quickbooks/APIHelper.php [hunk 2]
```diff
@@ -95,6 +106,11 @@ public static function getAccountingDataServiceObject($forRefreshToken = FALSE)
 
     $dataService = \QuickBooksOnline\API\DataService\DataService::Configure($dataServiceParams);
 
+    $dataService->setLogLocation($logLocation);
+    if (!$logActivated) {
+        $dataService->disableLog();
+    }
+
```
Comment (ananelson) [inline CRM/Quickbooks/APIHelper.php:113]: Must call these each time we create a dataService.

File: CRM/Civiquickbooks/Upgrader.php [hunk 3]
```diff
@@ -138,4 +138,13 @@ public function upgrade_20203() {
 
     return true;
   }
+
+  public function upgrade_20204() {
+    $this->ctx->log->info('Setting defaults on new settings for compatibility.');
+
+    civicrm_api3('Setting', 'create', [ 'quickbooks_log_dir' => '/tmp/IdsLogs' ]);
```
Comment (ananelson) [inline CRM/Civiquickbooks/Upgrader.php:145]: This is QBO's default value for this setting.

File: CRM/Civiquickbooks/Upgrader.php [hunk 4]
```diff
@@ -138,4 +138,13 @@ public function upgrade_20203() {
 
     return true;
   }
+
```
Comment (agileware-fj) [inline CRM/Civiquickbooks/Upgrader.php:141]: Is this upgrade function actually necessary?  I thought the Settings API returns the defaults for any setting that hasn't been added to the database...

Comment (ananelson) [reply CRM/Civiquickbooks/Upgrader.php:141]: That could be. Not the easiest for me to test right now but I'm happy to remove this if it's not needed.

**Comment by agileware-fj** [comment]
Tested fine, also I confirmed that the settings work without the database upgrade, so I've merged it with the upgrade code removed.

Thanks for submitting, @ananelson

**Pull request merged**